idea-plugin/src/main/kotlin/arrow/meta/ide/dsl/editor/intention/IntentionSyntax.kt

```diff
@@ -0,0 +1,159 @@
+package arrow.meta.ide.dsl.editor.intention
+
+import arrow.meta.ide.IdeMetaPlugin
+import arrow.meta.ide.phases.editor.intention.IntentionExtensionProvider
+import arrow.meta.internal.Noop
+import arrow.meta.phases.ExtensionPhase
+import com.intellij.codeInsight.intention.IntentionAction
+import com.intellij.codeInsight.intention.PriorityAction
+import com.intellij.codeInsight.intention.impl.config.IntentionActionMetaData
+import com.intellij.openapi.editor.Editor
+import org.jetbrains.kotlin.diagnostics.Diagnostic
+import org.jetbrains.kotlin.idea.intentions.SelfTargetingIntention
+import org.jetbrains.kotlin.idea.quickfix.KotlinIntentionActionsFactory
+import org.jetbrains.kotlin.idea.quickfix.KotlinSingleIntentionActionFactory
+import org.jetbrains.kotlin.idea.quickfix.QuickFixContributor
+import org.jetbrains.kotlin.psi.KtElement
+import com.intellij.openapi.actionSystem.AnAction
+
+/**
+ * The IDE analysis user code and provides [IntentionAction]'s to either signal error's to user's or resolve them if triggered.
+ */
+interface IntentionSyntax : IntentionUtilitySyntax {
+  // TODO: [addIntentionWithMetaData] Bails if there is no html and intentionDescription. Add MetaData in a function and not with through the resource folder.
```

Comment:
Should this be kept?

Reply:
Good call let's deprecate it until we have a stable version